src/System.IO.Pipelines/FlushAsyncAwaitable.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace System.IO.Pipelines
+{
+    public struct FlushAsyncAwaitable : ICriticalNotifyCompletion
```

Comment:
Some of these types are what I would call "plumbing". We should consider moving them to some obscure sub namespace. But let's chat about it when we do an API review.